src/kitovu/sync/filecache.py

```diff
@@ -0,0 +1,115 @@
+"""FileCache keeps track of the state of the files, remotely and locally. It exists to determine if the local file
+has been changed between two synchronisation processes and allows for a conflict handling accordingly."""
+
+from enum import Enum
+import json
+import pathlib
+import typing
+
+from kitovu.sync import syncplugin
+from kitovu import utils
+
+# sync-process:
+# every time we sync something, the FileCache gets read, upon which kitovu decides what needs to be downloaded.
+# The files get downloaded, for each file, the FileCache gets updated for the files that were synchronised.
+
+
+class Filestate(Enum):
+    """Used to discern in which places files have changed"""
+    LOCAL = 1
+    REMOTE = 2
+    BOTH = 3
+    NONE = 4
+
+
+# #FIXME refactor with attrs.?
+class File:
+
+    def __init__(self, local_digest_at_synctime: str,
+                 relative_path_with_filename: pathlib.Path,
+                 plugin: syncplugin.AbstractSyncPlugin) -> None:
+        self._cached_digest = local_digest_at_synctime  # dict-value
+        self._relative_path_with_filename: pathlib.Path = relative_path_with_filename  # dict-key
+        self._plugin = plugin  # dict-value
+        self._remote_digest = ""
+        self._local_digest = ""
+
+    def to_dict(self) -> typing.Dict[str, str]:
+        return {"plugin": self._plugin.name,  # FIXME doesn't resolve
+                "digest": self._cached_digest}
+
+
+class FileCache:
+
+    def __init__(self, filename: pathlib.Path):
+        self._filename: pathlib.Path = filename
+        self._data: typing.Dict[pathlib.Path, File] = {}
+
+    def _compare_digests(self, remote_digest: str, local_digest: str, cached_digest: str) -> Filestate:
+        local_changed: bool = local_digest != cached_digest
+        remote_changed: bool = remote_digest != cached_digest
```

Comment:
You could rewrite the `==` parts as well, using e.g. `elif remote_changed and not local_changed:` (case 4) or `if not remote_changed and not local_changed:` (case 5), etc.